src/rprblender/engine/viewport_engine.py

```diff
@@ -112,31 +112,36 @@ def __init__(self, context: bpy.types.Context):
         self.width, self.height = x2 - x1, y2 - y1
         self.border = (x1, y1), (self.width, self.height)
 
-        if scene.rpr.viewport_limits.limit_viewport_resolution and self.width * self.height > 0:
-            # changing render resolution in case of enabled limit_viewport_resolution property
-
-            render_w = int(scene.render.resolution_x * scene.render.resolution_percentage / 100.0)
-            render_h = int(scene.render.resolution_y * scene.render.resolution_percentage / 100.0)
-
-            region_aspect = self.width / self.height
-            render_aspect = render_w / render_h
-
-            if render_aspect > region_aspect:
-                # if render resolution is wider, use the max height
-                # from render and scale to aspect ratio
-                self.height = min(render_h, self.height)
-                self.width = int(region_aspect * self.height)
-            else:
-                # scale to render width and maintain aspect ratio
-                self.width = min(render_w, self.width)
-                self.height = int(self.width / region_aspect)
-
     def export_camera(self, rpr_camera):
         """Exports camera settings with render border"""
         self.camera_data.export(rpr_camera,
             ((self.border[0][0] / self.screen_width, self.border[0][1] / self.screen_height),
              (self.border[1][0] / self.screen_width, self.border[1][1] / self.screen_height)))
 
+    def adapt_resolution(self, requested_ratio, prev_settings, min_scale):
```

Comment:
One other thing I noticed here is this can actually scale UP the viewport yes?  I don't think we ever want to do the viewport more than the original resolution.  Might be worth putting in a check for that.

Reply:
Added check that render resoultion <= viewport resolution